Compatibility with Svelte 5

`svelte-icon` cannot be used in a Svelte 5 project due to an expected 4.0 Svelte dependency instead:
```
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: web@0.0.1
npm error Found: svelte@5.1.14
npm error node_modules/svelte
npm error   dev svelte@"^5.0.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer svelte@"^4.0.0" from @steeze-ui/svelte-icon@1.5.0
npm error node_modules/@steeze-ui/svelte-icon
npm error   dev @steeze-ui/svelte-icon@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

I would be nice to have compatibility with Svelte 5.

--- a/packages/components/svelte-icon/CHANGELOG.md
+++ b/packages/components/svelte-icon/CHANGELOG.md
@@ -1,5 +1,9 @@
 # @steeze-ui/svelte-icon
 
+## 1.6.0
+
+- add svelte 5 support
+
 ## 1.5.0
 
 ### Minor Changes

--- a/packages/components/svelte-icon/package.json
+++ b/packages/components/svelte-icon/package.json
@@ -19,7 +19,7 @@
 		".": {
 			"types": "./dist/index.d.ts",
 			"import": "./dist/index.js",
-			"svelte": "./dist/index.js"
+			"default": "./dist/index.js"
 		}
 	},
 	"files": [
@@ -34,7 +34,7 @@
 		"@steeze-ui/heroicons": "^2.4.0",
 		"@sveltejs/adapter-auto": "^2.1.0",
 		"@sveltejs/kit": "^2.0.0",
-		"@sveltejs/package": "^2.1.0",
+		"@sveltejs/package": "^2.3.4",
 		"@sveltejs/vite-plugin-svelte": "^4.0.1",
 		"prettier": "^3.2.2",
 		"prettier-plugin-svelte": "^3.2.8",

--- a/packages/components/svelte-icon/package.json
+++ b/packages/components/svelte-icon/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "@steeze-ui/svelte-icon",
-	"version": "1.5.0",
+	"version": "1.6.0",
 	"scripts": {
 		"dev": "vite dev",
 		"build": "npm run package",
@@ -28,21 +28,22 @@
 		"!dist/**/*.spec.*"
 	],
 	"peerDependencies": {
-		"svelte": "^4.0.0"
+		"svelte": "^5.0.0"
 	},
 	"devDependencies": {
-		"@steeze-ui/heroicons": "^2.2.2",
+		"@steeze-ui/heroicons": "^2.4.0",
 		"@sveltejs/adapter-auto": "^2.1.0",
-		"@sveltejs/kit": "^1.20.5",
+		"@sveltejs/kit": "^2.0.0",
 		"@sveltejs/package": "^2.1.0",
-		"prettier": "^2.8.8",
-		"prettier-plugin-svelte": "^2.10.1",
+		"@sveltejs/vite-plugin-svelte": "^4.0.1",
+		"prettier": "^3.2.2",
+		"prettier-plugin-svelte": "^3.2.8",
 		"publint": "^0.1.12",
-		"svelte": "^4.0.0",
-		"svelte-check": "^3.4.4",
+		"svelte": "^5.2.2",
+		"svelte-check": "^4.0.0",
 		"tslib": "^2.5.3",
 		"typescript": "^5.1.3",
-		"vite": "^4.3.9"
+		"vite": "^5.0.3"
 	},
 	"publishConfig": {
 		"access": "public",

--- a/packages/components/svelte-icon/src/lib/Icon.svelte
+++ b/packages/components/svelte-icon/src/lib/Icon.svelte
@@ -1,12 +1,15 @@
 <script lang="ts">
 	import type { IconSource } from './index.js'
 
-	export let src: IconSource
-	export let size = '100%'
-	export let theme = 'default'
-	export let title: string | undefined = undefined
+	let { src, size = '100%', theme = 'default', title = undefined, ...restProps } : {
+		src: IconSource,
+		size: string,
+		theme: string,
+		title: string | undefined,
+		[key: string]: any
+	} = $props()
 
-	$: icon = src?.[theme] ?? src?.['default'] ?? Object.values(src)?.[0]
+	const icon = $derived(src?.[theme] ?? src?.['default'] ?? Object.values(src)?.[0])
 
 	if (size !== '100%') {
 		if (size.slice(-1) != 'x' && size.slice(-1) != 'm' && size.slice(-1) != '%') {
@@ -19,7 +22,7 @@
 	}
 </script>
 
-<svg {...icon?.a} xmlns="http://www.w3.org/2000/svg" width={size} height={size} {...$$restProps}>
+<svg {...icon?.a} xmlns="http://www.w3.org/2000/svg" width={size} height={size} {...restProps}>
 	{#if title}
 		<title>{title}</title>
 	{/if}

--- a/packages/components/svelte-icon/src/routes/+page.svelte
+++ b/packages/components/svelte-icon/src/routes/+page.svelte
@@ -59,7 +59,7 @@
 								name="{iconpack}-theme"
 								value={theme}
 								{checked}
-								on:change={(e) => (iconpacks[iconpack].activeTheme = theme)}
+								onchange={(e) => (iconpacks[iconpack].activeTheme = theme)}
 							/>
 							<label for={theme} style="color:white">{theme}</label>
 						</div>

--- a/packages/components/svelte-icon/svelte.config.js
+++ b/packages/components/svelte-icon/svelte.config.js
@@ -1,5 +1,5 @@
 import adapter from '@sveltejs/adapter-auto'
-import { vitePreprocess } from '@sveltejs/kit/vite'
+import { vitePreprocess } from '@sveltejs/vite-plugin-svelte';
 
 /** @type {import('@sveltejs/kit').Config} */
 const config = {
@@ -16,3 +16,4 @@ const config = {
 }
 
 export default config
+

--- a/packages/components/svelte-icon/vite.config.ts
+++ b/packages/components/svelte-icon/vite.config.ts
@@ -1,8 +1,6 @@
-import { sveltekit } from '@sveltejs/kit/vite'
+import { sveltekit } from '@sveltejs/kit/vite';
+import { defineConfig } from 'vite';
 
-/** @type {import('vite').UserConfig} */
-const config = {
+export default defineConfig({
 	plugins: [sveltekit()]
-}
-
-export default config
+});

